addons/website/models/ir_http.py

```diff
@@ -250,7 +250,7 @@ def binary_content(self, xmlid=None, model='ir.attachment', id=None, field='data
             obj = self.env[model].browse(int(id))
         if obj and 'website_published' in obj._fields:
             if self.env[obj._name].sudo().search([('id', '=', obj.id), ('website_published', '=', True)]):
-                self.sudo()
+                self = self.sudo()
```

Comment:
@qsm-odoo 

so no, it was wrong

Reply:
Thanks. BTW, if you can explain to me why the if condition is not `obj.sudo().website_published`...

Reply:
cc @rdeodoo

Reply:
(Btw my previous message was written by JKE using my computer :P)

And yup as you say, the code should be
```python
if obj.sudo().website_published:
    self = self.sudo()
```